meta-client/src/main/java/org/apache/helix/metaclient/recipes/lock/LockInfo.java

```diff
@@ -0,0 +1,250 @@
+package org.apache.helix.metaclient.recipes.lock;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
+import org.apache.helix.zookeeper.datamodel.ZNRecord;
+
+
+/**
+ * This structure represents a Lock node information, implemented using ZNRecord
+ */
+public class LockInfo {
+
+  // Default values for each attribute if there are no current values set by user
+  public static final String DEFAULT_LOCK_ID_TEXT = "";
+  public static final String DEFAULT_OWNER_ID_TEXT = "";
+  public static final String DEFAULT_CLIENT_ID_TEXT = "";
+  public static final String DEFAULT_CLIENT_DATA = "";
+  public static final long DEFAULT_GRANTED_AT_LONG = -1L;
+  public static final long DEFAULT_LAST_RENEWED_AT_LONG = -1L;
+  public static final Duration DEFAULT_TIMEOUT_DURATION = Duration.ofMillis(-1L);
+  private static final String ZNODE_ID = "LOCK";
+  private final ZNRecord _record;
+
+  /**
+   * The keys to lock information
+   */
+  public enum LockInfoAttribute {
+    LOCK_ID,
+    OWNER_ID,
+    CLIENT_ID,
+    //CLIENT_DATA is a string that represents the serialized object passed in from the user.
+    CLIENT_DATA,
+    GRANTED_AT,
+    LAST_RENEWED_AT,
+    TIMEOUT
+  }
+
+  /**
+   * Initialize a default LockInfo instance
+   */
+  private LockInfo() {
+    _record = new ZNRecord(ZNODE_ID);
+    setLockInfoFields(DEFAULT_LOCK_ID_TEXT, DEFAULT_OWNER_ID_TEXT, DEFAULT_CLIENT_ID_TEXT, DEFAULT_CLIENT_DATA, DEFAULT_GRANTED_AT_LONG,
+        DEFAULT_LAST_RENEWED_AT_LONG, DEFAULT_TIMEOUT_DURATION);
+  }
+
+  /**
+   * Initialize a LockInfo with a ZNRecord, set all info fields to default data
+   * @param znRecord The ZNRecord contains lock node data that used to initialize the LockInfo
+   */
+  public LockInfo(ZNRecord znRecord) {
+    this();
```

Comment:
I don't quite understand. Do you mean change just the name of the variable instead of znrecord to something more generic or create a wrapper around znrecord?

Reply:
@xyuanlu I remember ZNRecord class is already used in some other MetaClient classes. @Marcosrico Can you confirm? If so, we can do a refactor later with a separate commit.

Reply:
Thanks for the clarification, fixed it. Please take a look, thanks!